src/main/java/edu/cornell/kfs/module/purap/document/web/struts/CuRequisitionAction.java

```diff
@@ -54,30 +54,29 @@
+    public ActionForward addItem(ActionMapping mapping, ActionForm form, HttpServletRequest request, 
+            HttpServletResponse response) throws Exception {
         PurchasingFormBase purchasingForm = (PurchasingFormBase) form;
         PurApItem item = purchasingForm.getNewPurchasingItemLine();
         RequisitionItem requisitionItem = (RequisitionItem) item;
         PurchasingDocument purDocument = (PurchasingDocument) purchasingForm.getDocument();
         
         if (StringUtils.isBlank(requisitionItem.getPurchasingCommodityCode())) {
-            boolean commCodeParam = SpringContext.getBean(ParameterService.class)
-                    .getParameterValueAsBoolean(CuRequisitionDocument.class, PurapParameterConstants.ENABLE_DEFAULT_VENDOR_COMMODITY_CODE_IND);
-
-            if (commCodeParam) {
-                if (purchasingForm instanceof RequisitionForm) {
-                    CuRequisitionDocument reqs = (CuRequisitionDocument) purchasingForm.getDocument();
-                    VendorDetail dtl = reqs.getVendorDetail();
-                    if (ObjectUtils.isNotNull(dtl)) {
-                        List<VendorCommodityCode> vcc = dtl.getVendorCommodities();
-                        String defaultCommodityCode = "";
-                        Iterator<VendorCommodityCode> it = vcc.iterator();
-                        while (it.hasNext()) {
-                            VendorCommodityCode commodity = it.next();
-                            if (commodity.isCommodityDefaultIndicator()) {
-                                defaultCommodityCode = commodity.getPurchasingCommodityCode();
-                                requisitionItem.setPurchasingCommodityCode(defaultCommodityCode);
-                            }
+            boolean commCodeParam = SpringContext.getBean(ParameterService.class).getParameterValueAsBoolean(
+                    RequisitionDocument.class, PurapParameterConstants.ENABLE_DEFAULT_VENDOR_COMMODITY_CODE_IND);
+
+            if (commCodeParam && purchasingForm instanceof RequisitionForm) {
+                CuRequisitionDocument reqs = (CuRequisitionDocument) purchasingForm.getDocument();
+                VendorDetail dtl = reqs.getVendorDetail();
+                if (ObjectUtils.isNotNull(dtl)) {
+                    List<VendorCommodityCode> vcc = dtl.getVendorCommodities();
+                    String defaultCommodityCode = "";
+                    Iterator<VendorCommodityCode> it = vcc.iterator();
+                    while (it.hasNext()) {
+                        VendorCommodityCode commodity = it.next();
+                        if (commodity.isCommodityDefaultIndicator()) {
+                            defaultCommodityCode = commodity.getPurchasingCommodityCode();
+                            requisitionItem.setPurchasingCommodityCode(defaultCommodityCode);

```

Comment:
This block may not have been cleaned up as much as it should have been in some previous PRs, since it still uses an Iterator and makes use of an unnecessary defaultCommodityCode variable. Should this be cleaned up as part of this PR? Or can we just defer that to a future user story?

Reply:
updated